config.py

```diff
@@ -742,157 +754,183 @@ def check_yaml(conf_path):
     for deprecated_config in deprecated_configs_enabled:
         msg = "Configuring %s in datadog.conf is not supported anymore. Please use conf.d" % deprecated_config
         deprecated_checks[deprecated_config] = {'error': msg, 'traceback': None}
         log.error(msg)
+    return deprecated_checks
 
-    osname = get_os()
-    checks_paths = [glob.glob(os.path.join(agentConfig['additional_checksd'], '*.py'))]
-
+def _all_configs_paths(osname, agentConfig):
+    """ Retrieve all configs and return their paths
+    """
     try:
-        checksd_path = get_checksd_path(osname)
-        checks_paths.append(glob.glob(os.path.join(checksd_path, '*.py')))
+        confd_path = get_confd_path(osname)
+        all_configs = [glob.glob(os.path.join(confd_path, '*.yaml'))]
     except PathNotFound, e:
-        log.error(e.args[0])
+        log.error("No conf.d folder found at '%s' or in the directory where the Agent is currently deployed.\n" % e.args[0])
         sys.exit(3)
 
+    # Compatibility code for the Nagios checks if it's still configured
+    # in datadog.conf
+    # FIXME: 6.x, should be removed
+    if not any('nagios' in config for config in itertools.chain(*all_configs)):
+        # check if it's configured in datadog.conf the old way
+        if any([nagios_key in agentConfig for nagios_key in NAGIOS_OLD_CONF_KEYS]):
+            all_configs.append('deprecated_nagios')
+
+
+    return all_configs
+
+def _checks_places(agentConfig, osname):
+    """ Return methods to generated paths to inspect for a check provided it's name
+    """
     try:
-        confd_path = get_confd_path(osname)
+        checksd_path = get_checksd_path(osname)
     except PathNotFound, e:
-        log.error("No conf.d folder found at '%s' or in the directory where the Agent is currently deployed.\n" % e.args[0])
+        log.error(e.args[0])
         sys.exit(3)
 
-    # We don't support old style configs anymore
-    # So we iterate over the files in the checks.d directory
-    # If there is a matching configuration file in the conf.d directory
-    # then we import the check
-    for check in itertools.chain(*checks_paths):
-        check_name = os.path.basename(check).split('.')[0]
-        check_config = None
-        if check_name in initialized_checks or check_name in init_failed_checks:
-            log.debug('Skipping check %s because it has already been loaded from another location', check)
-            continue
+    places = [lambda name: os.path.join(agentConfig['additional_checksd'], '%s.py' % name)]
 
-        # Let's see if there is a conf.d for this check
-        conf_path = os.path.join(confd_path, '%s.yaml' % check_name)
-        conf_exists = False
+    try:
+        thrird_party_path = get_3rd_party_path(osname)
+        places.append(lambda name: os.path.join(thrird_party_path, name, 'check.py'))
+    except PathNotFound:
+        pass
```

Comment:
good question, maybe a debug log as windows and macos won't have this 3rd party folder and we don't want to spam them for that